File: 0077-combinations/0077-combinations.java
class Solution {
    List<List<Integer>> ans = new ArrayList<>();
    public List<List<Integer>> combine(int n, int k) {
        recur(1, n, k, new ArrayList<>());
        return ans;
    }
    public void recur(int curr, int n, int k, List<Integer> list) {
        if(list.size() == k) {
            ans.add(new ArrayList<>(list));
            return;
        }
        if(curr > n) return;
        list.add(curr);
        recur(curr + 1, n, k, list);
        list.remove(list.size() - 1);
        recur(curr + 1, n, k, list);
    }
}

File: 0077-combinations/0077-combinations.cpp
class Solution {
public:
    vector<vector<int>> ans;
    vector<vector<int>> combine(int n, int k) {
        vector<int> temp;
        recur(1, n, k, temp);
        return ans;
    }
    void recur(int i, int n, int k, vector<int>& temp) {
        if(temp.size() == k) {
            ans.push_back(temp);
            return;
        }
        if(i > n) return;
        temp.push_back(i);
        recur(i + 1, n, k, temp);
        temp.pop_back();
        recur(i + 1, n , k, temp);
    }
};